notes/css.md

```diff
@@ -0,0 +1,248 @@
+де частини нижчого рангу представляють нижчі порядки:
+```
+[ inline ][ id ][ class/attribute/pseudo-class ][ type/pseudo-element ]
+```
+
+Наприклад, для селектора
+`span.visually.hidden:not(:focus, :focus-within)` —
+`[0][0][3][1] = 31` (`:not()` має специфічніть того, що всередині).
+
+### !important
+
+Властивості позначені `!important` кладуться в **окрему** пріоритетну чергу.
+
+Специфічніть в пріоритетній черзі поміж елементами визначається таким же чином, але всі _важливі_ властивості будуть мати вищу специфічність, ніж будь-який селектор в нормальній черзі.
+
+
+### `:is()`, `:where()` — комбінованi селектор
+
+Дозволяє комбінувати довгі селектори зручніше.
+Для `:is()` специфічність рахується так само, ніби вони були розвернуті без `:is()`.
+
+Наприклад, селектор довільної секції, яка йде зразу після довільної секції:
+
+```css
+:is(article, aside, section, nav) + :is(article, aside, section, nav) {
+	margin-block-start: 1rem;
+}
+```
+
+Це аналогічно до перебору усіх комбінацій:
+```css
+article + article,
+article + aside,
+article + section,
+article + nav,
+
+aside + article,
+/* ... 11 селекторів опущено */
+nav + nav {
+	margin-block-start: 1rem;
+}
+```
+
+`:where()` працює аналогічно, але має специфічність 0.
+
+
+### `:has()`
+
+Дозволяє стилізувати елемент, який має певні дочірні або сусідні елементи, які відповідають селектору всередині.
+
+`article:has(h2)` — це така стаття, всередині якої є заголовок другого рівня.
+
+
+### Псевдо-елементи
+
+`::before` та `::after` для того, щоб можна було дадавати _презентаційний_ контент елементам —
+такий, що не несе смислового навантаження, ні візуального, ні текстового.
+Якщо без контенту в них не можна обійтися, або треба, щоб він копіювався,
+цей контент має бути в розмітці, а не в стилях.
+
+`::before` та `::after` не можна виділити чи скопіювати, їх немає в [replaced elements](https://developer.mozilla.org/docs/Web/CSS/Replaced_element) (`img`, `input`); вони додаються тільки коли є `content`, і стандартно мають `display: inline`.
+
+Інші псевдо-елементи додволяють стилізувати те, чого явно немає в дереві документу —
+виділення (`::selection`),
+маркери списків (`::marker`),
+текст-підказка (`::placeholder`),
+першу літеру чи рядок (`::first-letter`, `::first-line`).
+
+[Список псевдо-елементів на MDN](https://developer.mozilla.org/docs/Web/CSS/Pseudo-elements)
+
+### Псевдо-класи
+
+Для того, щоб можна було доступитися до специфісних станів елементів, наприклад, `:visited`, `:invalid`.
+
+[Список псевдо-класів на MDN](https://developer.mozilla.org/docs/Web/CSS/Pseudo-classes)
+
+
+Values and Units
+-
+Використовуйте `rem` та `em` для віступів та полів (`margin`, `padding`, `gap` тощо).
+Використовуйте ці та інші відносні до розміру шрифту величини — `ch`, `lh`, `ex`, `cap`, а також відсотки (`%`) для встановлення розмірів елементів, де це необхідно.
+Це робить дизайн гнучким до налаштувань шрифту **користувача**.
+
+Використовуйте <b>новий синтаксис</b> для написання кольорів:
```

Comment:
```suggestion
Використовуйте **новий синтаксис** для написання кольорів:
```
What about adding browser support?

Reply:
I find that ones who are interested could find it on the linked pages but I am open to suggestions.
Also, there is a preset-env linked in the bottom of the file.

To be fair, `lh` and `cap` are supported nowhere, so ones who use it, will find it doesn't work pretty quickly 😄